### ISSUE TITLE
LSTMBlockFusedCell  does not support using  DropoutWrapper 

### ISSUE THREAD

**Issue by hanyh** [tensorflow/tensorflow#13649]
I am trying to use DropoutWrapper with LSTMBlockFusedCell as follows:

```
cell = tf.contrib.rnn.LSTMBlockFusedCell(num_units,forget_bias) 
cell = tf.contrib.rnn.DropoutWrapper(cell,dropout)
```
 I get an exception that the LSTMBlockFusedCell is not an RNNCell
Message: The parameter cell is not a RNNCell. Which is raised form _like_rnncell during DropoutWrapper initialization. 

It is checking for those proprieties on the cell:
""Checks that a given object is an RNNCell by using duck typing."""
   conditions = [hasattr(cell, "output_size"), hasattr(cell, "state_size"),                 hasattr(cell, "zero_state"), callable(cell)]  LSTMBlockFusedCell does not have output_size , state_size or zero_state properties. 

Should LSTMBlockFusedCell  act like RNNCell to allow using various wrappers?

https://stackoverflow.com/questions/46699985/using-tensorflow-dropoutwrapper-with-lstmblockfusedcell

**Comment by reedwm** [comment]
@ebrevdo any plans to support this?

**Comment by ebrevdo** [comment]
Unfortunately, because this is cell is fused in time, adding dropout would require modifying the C++ code that performs the computation to add proper random number generation and dropout.  Work in this direction is not planned at the moment but I'll ping @ekelsen  to get his thoughts on feasibility.

**Comment by xiaoyunwu** [comment]
why? the http://arxiv.org/abs/1409.2329 says the dropout is only applied to input not state, why can we just do a dropout on the input first?

**Comment by ebrevdo** [comment]
You're right.  This should be sufficiently easy, by adding new
input_keep_prob and variational_recurrent args to the constructor.  If
someone is interested in adding this, I'm happy to review the PR.

**Comment by facaiy** [comment]
@ebrevdo Do you mean to implement dropout in `LSTMBlockFusedCell`, rather than make `DropoutWrapper` supports `LSTMBlockFusedCell`, right?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by ebrevdo** [comment]
Yes, that's what I mean.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activityand the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by facaiy** [comment]
@hanyh @xiaoyunwu Hi, we found that user can call `tf.nn.dropout` on their inputs directly (see discussion with @ebrevdo in #15843), . Does the workaround solve your problem?

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by ebrevdo** [comment]
Assuming my solution is sufficient. Reopen if you find another reason to move dropout into this layer.

**Comment by godefv** [comment]
What about MultiRNNCell ? I believe this was the whole point of DropoutWraper, so that higher level functions accepting rnn cells can be used.

**Comment by hanyh** [comment]
I agree with @godefv  the workaround is ok for dropout wrapper, but you cannot still use it MultiRNNCell and dynamic_rnn
